pkg/volume/iscsi/iscsi_util.go

```diff
@@ -99,12 +100,22 @@ func (util *ISCSIUtil) MakeGlobalPDName(iscsi iscsiDisk) string {
 
 func (util *ISCSIUtil) AttachDisk(b iscsiDiskMounter) error {
 	var devicePath string
-	if b.iface == "default" {
+	var iscsiTransport string
+
+	out, err := b.plugin.execCommand("iscsiadm", []string{"-m", "iface", "-I", b.iface, "-o", "show"})
+	if err != nil {
+		glog.Errorf("iscsi: could not read iface %s error: %s", b.iface, string(out))
+		return err
```

Comment:
add the above log info to `err` so we know where the problem comes from in kubectl output.